client/modules/User/pages/SignupView.jsx

```diff
@@ -71,21 +87,29 @@ function mapDispatchToProps(dispatch) {
 }
 
 function asyncValidate(formProps, dispatch, props) {
-  const fieldToValidate = props.form._active;
-  if (fieldToValidate) {
-    const queryParams = {};
-    queryParams[fieldToValidate] = formProps[fieldToValidate];
-    queryParams.check_type = fieldToValidate;
-    return axios.get('/api/signup/duplicate_check', { params: queryParams })
-      .then((response) => {
-        if (response.data.exists) {
-          const error = {};
-          error[fieldToValidate] = response.data.message;
-          throw error;
-        }
-      });
-  }
-  return Promise.resolve(true).then(() => {});
+  const errors = {};
+  return Promise.resolve(true)
```

Comment:
is there a better way to write this code? the promises are nested here, and usually with promises it makes sense to have them only one level deep, if that make sense.